File: LISTAS/LISTA ENLAZADA/LISTAENLAZADA.js
class Nodo {
    constructor(dato) {
        this.dato = dato;
        this.siguiente = null;
    }
}

class ListaEnlazada {
    constructor() {
        this.cabeza = null;
    }
    
    insertarInicio(dato) {
        const nuevo = new Nodo(dato);
        nuevo.siguiente = this.cabeza;
        this.cabeza = nuevo;
    }
    
    insertarFinal(dato) {
        const nuevo = new Nodo(dato);
        if (!this.cabeza) {
            this.cabeza = nuevo;
            return;
        }
        
        let actual = this.cabeza;
        while (actual.siguiente) {
            actual = actual.siguiente;
        }
        actual.siguiente = nuevo;
    }
    
    eliminar(dato) {
        if (!this.cabeza) return;
        
        if (this.cabeza.dato === dato) {
            this.cabeza = this.cabeza.siguiente;
            return;
        }
        
        let actual = this.cabeza;
        while (actual.siguiente && actual.siguiente.dato !== dato) {
            actual = actual.siguiente;
        }
        
        if (actual.siguiente) {
            actual.siguiente = actual.siguiente.siguiente;
        }
    }
    
    mostrar() {
        const elementos = [];
        let actual = this.cabeza;
        while (actual) {
            elementos.push(actual.dato);
            actual = actual.siguiente;
        }
        return elementos;
    }
}

File: LISTAS/LISTA ENLAZADA/LISTAENLAZADA.py
class Nodo:
    def __init__(self, dato):
        self.dato = dato
        self.siguiente = None

class ListaEnlazada:
    def __init__(self):
        self.cabeza = None
    
    def insertar_inicio(self, dato):
        nuevo = Nodo(dato)
        nuevo.siguiente = self.cabeza
        self.cabeza = nuevo
    
    def insertar_final(self, dato):
        nuevo = Nodo(dato)
        if not self.cabeza:
            self.cabeza = nuevo
            return
        
        actual = self.cabeza
        while actual.siguiente:
            actual = actual.siguiente
        actual.siguiente = nuevo
    
    def eliminar(self, dato):
        if not self.cabeza:
            return
        
        if self.cabeza.dato == dato:
            self.cabeza = self.cabeza.siguiente
            return
        
        actual = self.cabeza
        while actual.siguiente and actual.siguiente.dato != dato:
            actual = actual.siguiente
        
        if actual.siguiente:
            actual.siguiente = actual.siguiente.siguiente
    
    def mostrar(self):
        elementos = []
        actual = self.cabeza
        while actual:
            elementos.append(actual.dato)
            actual = actual.siguiente
        return elementos